### DISCUSSION TITLE
Q: How to convert ToDo item to Project?

### DISCUSSION THREAD

**Discussion by ludwa6** [NicolasPetton/Projectify#22] [Q&A]
This is one of the most common errors made in GTD, especially with such an agile tool as Projectify: you Alt-C chuck a "simple" task into your Inbox, then quickly realize on review that there are some prerequisite subtasks -which means that this parent item is by definition a Project. 

So i'd like to know: what is the best practice for doing this little reassignment workflow?  I've tried what seemed most intuitive, which was to select this item in my Inbox, click the "Add subtasks" button... And that serves to add a "SubProject" tag to the parent item, alongside the "Inbox" and "todo" tags, but that does not make it show up in my Projects bin.

In another case: I went to another Inbox item with subtasks as yet undefined, threw it into my Projects bin... And that one now appears as a project, with the first "todo" task in its list being -itself.  I click on that task line, and nothing happens, because i'm already in that tiddler. 

I'd like to have one standard way of handling this quite common UseCase, that i can then train myself to do without having to think about it.  Any advice about this would be most welcome!

**Comment by NicolasPetton** [comment]
> This is one of the most common errors made in GTD, especially with such an agile tool as Projectify: you Alt-C chuck a "simple" task into your Inbox, then quickly realize on review that there are some prerequisite subtasks -which means that this parent item is by definition a Project.

For now the simplest way to do that is to remove the `todo` tag with the `Project` tag, there's no dedicated interface to convert a todo into a project (though I just added that to the roadmap!).

> In another case: I went to another Inbox item with subtasks as yet undefined, threw it into my Projects bin... And that one now appears as a project, with the first "todo" task in its list being -itself. I click on that task line, and nothing happens, because i'm already in that tiddler.

What's the "Projects bin"?

**Reply by ludwa6** [reply]
> What's the "Projects bin"?

Sorry: in terms of canonical names, i'm talking about the Dashboard, under Projects tab, where individual projects are viewed as Cards.

**Comment by NicolasPetton** [comment]
@ludwa6 would something like the following work for you for converting todos into projects?

![Screenshot from 2021-01-11 19-40-56](https://user-images.githubusercontent.com/123539/104224255-077c5f00-5445-11eb-9bd2-4c9842cd06d3.png)

**Comment by ludwa6** [comment]
> @ludwa6 would something like the following work for you for converting todos into projects?

That would certainly work, @NicolasPetton ... 

And the other way of coming at it is via the "Add Subprojects" button that sits beside the "Schedule" button on the default view of an "Inbox" "todo" tiddler.  If one clicks that button and adds a subtask, should that not convert the Parent tiddler into a Project?  That would be automatic, if we were playing strictly by the "rules" of GTD... But since not everyone does, maybe that would have to be made a conscious choice.  What do you think?

**Reply by NicolasPetton** [reply]
> But since not everyone does, maybe that would have to be made a conscious choice. What do you think?

Because not everyone uses GTD, I like the flexibility of having sub-projects (and nesting as much as I want).

That could be an setting though.

**Reply by NicolasPetton** [reply]
Regarding the "Add sub-tasks" button, I think I'll split it into a dropdown with 2 options:
- Add sub-tasks (this will be as today, it will make the todo into a sub-project)
- Convert into a sub-project.

Something like the following:
![Screenshot from 2021-01-11 21-42-07](https://user-images.githubusercontent.com/123539/104235863-e07a5900-5455-11eb-8dbe-490184fa4538.png)

**Comment by NicolasPetton** [comment]
Support was added in commit 75b169d.

**Comment by ludwa6** [comment]
Quick work, @NicolasPetton -nice!

Was not sure from what you've posted above if this new feature was added into v0.10... But i did the upgrade this morning anyway, and have been working it intensively for an hour.  I don't seem to have that "Convert into project" option, AFAICT, so i guess this is going into the next version -yes? 

Anyway: all good w/ the new version (that scheduling bug has not reappeared, b/t/w; yay!)... Only i still don't quite understand exactly how this is supposed to work.  So far it seems: If i add a Subtask to an Inbox Task, then the parent tiddler receives a tag of "SubProject", and then its Inbox appearance is altered by addition of a blue bubble after its title -which, if ticked, archives the "Project" (i only know this cuz i ticked it by accident, looked up the tiddler under "Recent" tab, and clicked the option to "Unarchive the Project.  Tooltip (hover-over) on this bubble this would be nice).  So in this sense, it is a Project... But if i want it to appear in the Projects section of my Dashboard, i have to add a Project tag to it.

Well: i guess that's what is coming w/ your proposed "Convert into project" option, which would be a helpful affordance.  But in my (GTD) taxonomy, a Project is a defined goal that has any number >1 of steps (i.e. Tasks) to complete it -so it would be great if this could be like a business rule that gets enforced by virtue of activating a "GTD-rules" setting, whereby adding a subtask to any task automatically makes it into a full-fledged Project.  My €0.02, FWIW :-)

**Reply by NicolasPetton** [reply]
The new "Convert into project" option It will indeed be coming in the next release.

> Well: i guess that's what is coming w/ your proposed "Convert into project" option, which would be a helpful affordance. But in my (GTD) taxonomy, a Project is a defined goal that has any number >1 of steps (i.e. Tasks) to complete it -so it would be great if this could be like a business rule that gets enforced by virtue of activating a "GTD-rules" setting, whereby adding a subtask to any task automatically makes it into a full-fledged Project. My €0.02, FWIW :-)

I don't want to restrict possible workflows, that's why you will have both options: "Convert into project" to make it into its own (top-level) project, and "Add sub-tasks" to keep the todo inside its existing project.

Adding workflow settings would be really nice, but that would be quite some work so I'm not planning to add that in the near future, unless more reasons to add such a setting arise.

**Reply by ludwa6** [reply]
OK @NicolasPetton : I will adapt my own workflow to these realities -no great hardship, after all!
